references.go

```diff
@@ -159,84 +155,28 @@ func blobHash(path string, commit *object.Commit) (hash plumbing.Hash, found boo
 	return file.Hash, true
 }
 
-type contentsComparatorFn func(path string, a, b *object.Commit) (bool, error)
-
 // Returns a new slice of commits, with duplicates removed.  Expects a
-// sorted commit list.  Duplication is defined according to "comp".  It
-// will always keep the first commit of a series of duplicated commits.
-func removeComp(path string, cs []*object.Commit, comp contentsComparatorFn) ([]*object.Commit, error) {
+// sorted commit list.  Duplication is defined according to the blob hash.
+// It will always keep the first commit of a series of duplicated commits.
+func removeRedundant(path string, cs []*object.Commit) ([]*object.Commit, error) {
 	result := make([]*object.Commit, 0, len(cs))
 	if len(cs) == 0 {
```

Comment:
I think you should add this at the start of this function and then remove the check here (saves extra unnecessary work):
```
	if len(cs) <= 1 {
		return cs, nil
	}
```